### PR TITLE
chore: move faker into own file

### PR DESCRIPTION
**Pull request by Shinigami92**
https://github.com/faker-js/faker/issues/405#issuecomment-1048613530

### CODE REVIEW

**Comment by Shinigami92** [comment]
@ST-DDT Could you please have a look if this is what you expected?
Also please have a look and jump into this PR if anything is needed for documentation (automation script and so on)

File: src/faker.ts [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    this.locale = locale;
+  }
+}
+
+// since we are requiring the top level of faker, load all locales by default
+export const faker: Faker = new Faker({
+  locales: allLocales,
+});
+
+export default faker;
```
Comment (ST-DDT) [inline src/faker.ts:152]: This should remain in the index, as this loads all the locales.

**Comment by ST-DDT** [comment]
> Also please have a look and jump into this PR if anything is needed for documentation (automation script and so on)

Will do so this evening.

File: src/faker.ts (hunk 1 above)
Comment (Shinigami92) [reply src/faker.ts:152]: :eyes: 
I start to not understand..., so maybe we should talk via voice about this when you have time.

**Comment by Shinigami92** [comment]
Please ignore the `codecov/patch` job for this PR

**Review by ST-DDT — APPROVED** [review]

**Review by JessicaSachs — CHANGES_REQUESTED** [review]
This doesn't fix the size issues yet. `pnpm build` for a Vite project that only does 

```js
import faker from '@faker-js/faker/locale/en'
console.log(faker.locales)
```

is still the same bundle size as a project that loads all of Faker. See pics of before + after sizes.

Screen Shot 2022-02-24 at 1 34 55 PM
Screen Shot 2022-02-24 at 1 35 16 PM

**Comment by Shinigami92** [comment]
Please have a look into the `dist/esm` folder and lock how many chunks are generated

**Review by JessicaSachs — APPROVED** [review]
Ahhh. Had to rerun. It works perfectly. Down 500kb minified + gzipped.
Screen Shot 2022-02-24 at 1 50 38 PM
.

**Pull request merged**